### PR TITLE
Fix some more warnings.

### PR DESCRIPTION
**Pull request by neheb**
clang's -Weverything seems to catch a lot of stuff. Here's some of it.

### CODE REVIEW

File: src/hashcat.c [hunk 1]
```diff
@@ -2895,7 +2896,7 @@ static int run_kernel (const uint kern_run, hc_device_param_t *device_param, con
     {
       if (data.opti_type & OPTI_TYPE_SLOW_HASH_SIMD)
       {
-        num_elements = CEIL ((float) num_elements / device_param->vector_width);
+        num_elements = CEIL (num_elements / device_param->vector_width);
```
Comment (magnumripper) [inline src/hashcat.c:2899]: Are you sure about this?

Comment (neheb) [reply src/hashcat.c:2899]: Briefly looking at it, both variables seem to be uints.

Comment (magnumripper) [reply src/hashcat.c:2899]: Hence the need for the typecast you removed. I'm not saying you are wrong, I'm just asking if you actually understand what you are doing here.

Comment (neheb) [reply src/hashcat.c:2899]: Not really. I added a printf to see if anything changed after I removed the cast. Numbers were still the same.

The only issue I can potentially see is that dividing could cause a remainder to get thrown away. num_elements seems to end up being a multiple of 8 though.

Comment (jsteube) [reply src/hashcat.c:2899]: should be fine, thanks for cross check @magnumripper

**Pull request merged**